store/badger.go

```diff
@@ -1,12 +1,20 @@
 package store
 
 import (
+	"context"
 	"errors"
 	"path/filepath"
+	"time"
 
 	"github.com/dymensionxyz/gerr-cosmos/gerrc"
 
-	"github.com/dgraph-io/badger/v3"
+	"github.com/dgraph-io/badger/v4"
+	"github.com/dgraph-io/badger/v4/options"
+)
+
+const (
+	gcTimeout    = 1 * time.Minute
+	discardRatio = 0.5
```

Comment:
could you comment a justification for these? link to some relevant forum thread or reasoning?

Reply:
done